templates/textinput.jsx

```diff
@@ -26,71 +27,88 @@ export default function TextInput (props) {
+                  <label
+                    className="textinput-item__prefix"
+                    id={`${_id}-${index}-aria`}
+                    htmlFor={`${_id}-${index}`}
+                    aria-label={prefix}
+                    dangerouslySetInnerHTML={{ __html: compile(prefix, props) }}
+                  >
+                  </label>
                 </div>
-                <div className="textinput-item__icon textinput-item__incorrect-icon" aria-label={_globals._accessibility._ariaLabels.incorrect}>
-                  <div className="icon" aria-hidden="true" />
+              }
+
+              <div className="textinput-item__textbox-container">
+                <input
+                  className="textinput-item__textbox js-textinput-textbox"
+                  type="text"
+                  placeholder={placeholder}
+                  data-id={`${input}-${index}`}
+                  id={`${_id}-${index}`}
+                  aria-labelledby={prefix && `${_id}-${index}-aria`}
+                  aria-label={placeholder}
+                  defaultValue={userAnswer}
+                  disabled={!_isEnabled}
+                />
+                <div className="textinput-item__state">
+                  <div className="textinput-item__icon textinput-item__correct-icon" aria-label={_globals._accessibility._ariaLabels.correct}>
+                    <div className="icon" aria-hidden="true"/>
+                  </div>
+                  <div className="textinput-item__icon textinput-item__incorrect-icon" aria-label={_globals._accessibility._ariaLabels.incorrect}>
+                    <div className="icon" aria-hidden="true" />
+                  </div>
                 </div>
               </div>
+
+              {suffix &&
+                <div className="textinput-item__suffix-container">
+                  <label
+                    className="textinput-item__suffix"
+                    id={`${_id}-${index}-aria`}
+                    htmlFor={`${_id}-${index}`}
+                    aria-label={suffix}
+                    dangerouslySetInnerHTML={{ __html: compile(suffix, props) }}
+                  >
+                  </label>
+                </div>
+              }
+
             </div>
 
-            {suffix &&
-              <div className="textinput-item__suffix-container">
-                <label
-                  className="textinput-item__suffix"
-                  id={`${_id}-${index}-aria`}
-                  htmlFor={`${_id}-${index}`}
-                  aria-label={suffix}
-                  dangerouslySetInnerHTML={{ __html: compile(suffix, props) }}
-                >
-                </label>
-              </div>
+            {_canShowCorrectness &&
+            <div
+              className="textinput-item__answer-container"
+              dangerouslySetInnerHTML={{
+                __html: (_isInteractionComplete && _canShowCorrectness && _correctAnswers) || '&nbsp;'
```

Comment:
Minor: We are already in a _canShowCorrectness condition, so the extra check is not needed here